src/ble/CHIPBleServiceData.h

```diff
@@ -46,11 +46,17 @@ enum chipBLEServiceDataType
 struct ChipBLEDeviceIdentificationInfo
 {
     constexpr static uint16_t kDiscriminatorMask = 0xfff;
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+    constexpr static uint8_t kAdditionalDataFlagMask = 0x1;
+#endif
 
     uint8_t OpCode;
     uint8_t DeviceDiscriminator[2];
     uint8_t DeviceVendorId[2];
     uint8_t DeviceProductId[2];
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
```

Comment:
This should be always present in the advertising data. It should only be set differently depending on whether the C3 characteristic is present or not. Therefore, the ifdefs can be removed in this header.

Reply:
@Damian-Nordic, I removed the ifdef. Now, for Linux and ESP32, if config option "CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING" is enabled, we set the Additional Data Flag and start showing C3 characteristic.

I also increased bumped the static assert check for size of BLE advertisement data in src/platform/Zephyr/BLEManagerImpl.cpp, PTAL!